From df8a2f6dc114b2c5c7685a069f717f2b06186b74 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Wed, 20 Nov 2024 16:23:06 +0000
Subject: [PATCH] rp1-pio: Add missing 'static inline's

Avoid some duplicate symbol errors by adding some missing
'static inline' decorations.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 include/linux/pio_rp1.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/include/linux/pio_rp1.h
+++ b/include/linux/pio_rp1.h
@@ -247,7 +247,7 @@ static inline bool pio_can_add_program_a
 	return !rp1_pio_can_add_program(client, &args);
 }
 
-uint pio_add_program(struct rp1_pio_client *client, const pio_program_t *program)
+static inline uint pio_add_program(struct rp1_pio_client *client, const pio_program_t *program)
 {
 	struct rp1_pio_add_program_args args;
 	int offset;
@@ -367,7 +367,7 @@ static inline int pio_sm_set_config(stru
 	return rp1_pio_sm_set_config(client, &args);
 }
 
-int pio_sm_exec(struct rp1_pio_client *client, uint sm, uint instr)
+static inline int pio_sm_exec(struct rp1_pio_client *client, uint sm, uint instr)
 {
 	struct rp1_pio_sm_exec_args args = { .sm = sm, .instr = instr, .blocking = false };
 
@@ -377,7 +377,7 @@ int pio_sm_exec(struct rp1_pio_client *c
 	return rp1_pio_sm_exec(client, &args);
 }
 
-int pio_sm_exec_wait_blocking(struct rp1_pio_client *client, uint sm, uint instr)
+static inline int pio_sm_exec_wait_blocking(struct rp1_pio_client *client, uint sm, uint instr)
 {
 	struct rp1_pio_sm_exec_args args = { .sm = sm, .instr = instr, .blocking = true };
 
